timeseries/src/autogluon/timeseries/splitter.py

```diff
@@ -172,11 +171,11 @@ def _split(
             can_be_split = item_index[long_enough]
             cannot_be_split = item_index[~long_enough]
 
-            train_dataframes.append(ts_dataframe.loc[cannot_be_split])
+            train_dataframes.append(ts_dataframe.query("item_id in @cannot_be_split"))
```

Comment:
Interesting. What's the speedup factor we get from this change and why was loc slow? Does it sort by index values in index lookups? and does query guarantee not losing original item order?

Reply:
`loc[pd.Index]` / `loc[list]` is extremely slow even for `pd.DataFrame`, not only for `TimeSeriesDataFrame`. 

For a DF with 7.5M rows:
- `loc` takes 23s
- `query` takes 0.9s

From my experience + description in the docs it seems that `query` should preserve order. According to the docs, it creates a boolean index internally and selects the rows that meet the condition this way.

Reply:
OK but we're also removing `.loc[original_item_order]` at the end of the method. Is the overall item order still preserved?